rasa_nlu/server.py

```diff
@@ -48,74 +46,117 @@ def requires_auth(f):
+        return json.dumps({'version': __version__})
+
+    @app.route("/config", methods=['GET'])
     @requires_auth
-    def rasaconfig():
-        return jsonify(config.as_dict())
+    def rasaconfig(self, request):
+        """Returns the in-memory configuration of the RASA server"""
+
+        request.setHeader('Content-Type', 'application/json')
+        return json.dumps(self.config.as_dict())
 
-    @rasa_nlu_app.route("/status", methods=['GET'])
+    @app.route("/status", methods=['GET'])
     @requires_auth
-    def status():
-        return jsonify(current_app.data_router.get_status())
+    def status(self, request):
+        """Returns the agents status"""
 
-    @rasa_nlu_app.route("/", methods=['GET'])
-    def hello():
-        return "hello from Rasa NLU: " + __version__
+        request.setHeader('Content-Type', 'application/json')
+        return json.dumps(self.data_router.get_status())
 
-    @rasa_nlu_app.route("/train", methods=['POST'])
+    @app.route("/train", methods=['POST'])
     @requires_auth
-    def train():
-        data_string = request.get_data(as_text=True)
-        current_app.data_router.start_train_process(data_string, request.args)
-        return jsonify(info="training started.", training_process_ids=current_app.data_router.train_proc_ids())
+    def train(self, request):
+        """Process http request containing user data to train a new/existing agent"""
+
+        def errback(f):
+            f.trap(ValueError)
+            logger.debug("error: {}".format(f.getErrorMessage()))
+
+        data_string = request.content.read().decode('utf-8', 'strict')
+
+        test = self.data_router.start_train_process(data_string,
+                                                    {key.decode('utf-8', 'strict'): value[0].decode('utf-8', 'strict')
+                                                     for key, value in request.args.items()})
 
-    logging.basicConfig(filename=config['log_file'], level=config['log_level'])
-    logging.captureWarnings(True)
-    logger.info("Configuration: " + config.view())
+        test.addCallback(lambda model_path: logger.info("Created model at: {}".format(model_path)))
+        test.addErrback(errback)
 
-    logger.debug("Creating a new data router")
-    rasa_nlu_app.data_router = DataRouter(config, component_builder)
-    return rasa_nlu_app
+        request.setHeader('Content-Type', 'application/json')
+        return json.dumps({"info": "Training started."})
 
 
 if __name__ == '__main__':
     # Running as standalone python application
     arg_parser = create_argparser()
     cmdline_args = {key: val for key, val in list(vars(arg_parser.parse_args()).items()) if val is not None}
     rasa_nlu_config = RasaNLUConfig(cmdline_args.get("config"), os.environ, cmdline_args)
-    app = WSGIServer(('0.0.0.0', rasa_nlu_config['port']), create_app(rasa_nlu_config))
+    rasa = RasaNLU(rasa_nlu_config)
+    rasa.app.run('0.0.0.0', rasa_nlu_config['port'])
```

Comment:
Since there is no `serve_forever` anymore, is this call blocking? If so, it would be helpful to move the following logger call before the run call as it contains the port information